src/executeCommandLine/executeCommandLine.ts

```diff
@@ -765,12 +764,38 @@ namespace ts {
             createReportErrorSummary(sys, buildOptions)
         );
         updateSolutionBuilderHost(sys, cb, buildHost);
+        enableSolutionPerformance(sys, buildOptions);
         const builder = createSolutionBuilder(buildHost, projects, buildOptions);
         const exitStatus = buildOptions.clean ? builder.clean() : builder.build();
+        reportSolutionBuilderTimes(sys, buildOptions, builder);
         dumpTracingLegend(); // Will no-op if there hasn't been any tracing
         return sys.exit(exitStatus);
     }
 
+    function enableSolutionPerformance(system: System, options: BuildOptions) {
+        if (system === sys && options.extendedDiagnostics) buildPerformance.enable();
+    }
+
+    function reportSolutionBuilderTimes(system: System, buildOptions: BuildOptions, builder: SolutionBuilder<BuilderProgram>) {
+        if (system !== sys || !buildOptions.extendedDiagnostics) return;
+
+        if (!buildPerformance.isEnabled()) {
+            sys.write(Diagnostics.Performance_timings_for_diagnostics_or_extendedDiagnostics_are_not_available_in_this_session_A_native_implementation_of_the_Web_Performance_API_could_not_be_found.message + "\n");
+            return;
+        }
+
+        const statistics: Statistic[] = [];
+        statistics.push(
+            { name: "Projects in scope", value: getBuildOrderFromAnyBuildOrder(builder.getBuildOrder()).length, type: StatisticType.count },
+        );
+        buildPerformance.forEachCount((count, name) => statistics.push({ name, value: count, type: StatisticType.count }));
+        buildPerformance.forEachStatistics(s => statistics.push(s));
+        buildPerformance.forEachMeasure((duration, name) => statistics.push({ name: `${name} time`, value: duration, type: StatisticType.time }));
+        buildPerformance.disable();
+        buildPerformance.enable();

```

Comment:
Rather than introducing a separate `buildPerformance` object, couldn't we just prefix builder-specific marks and measures and use that prefix as a filter when printing? If it's about resetting marks and measures between builds in watch mode, I'd rather we implement `clearMarks(name)` and `clearMeasures(name)` from the WHATWG User timings API:

- `clearMarks(name?)` - https://developer.mozilla.org/en-US/docs/Web/API/Performance/clearMarks
- `clearMeasures(name?)` - https://developer.mozilla.org/en-US/docs/Web/API/Performance/clearMeasures